### PR TITLE
Set a time limit on exchangeShare and add a method to get error information

### PR DESCRIPTION
**Pull request by KotaTakahashi9320**
# Summary
Set a time limit on exchangeShare
Add a method to get error information

QuickMPC-libClient-py 
https://github.com/acompany-develop/QuickMPC-libClient-py/pull/32
# Purpose
Server continues to wait even if client is terminating with an error
Since grpc determines that metadata is limited to 7kb, and stack trace information may exceed 7kb
# Contents
Fix getShare to wait up to 50 seconds
Add method to get error information
If the metadata exceeds 7kb, send a message recommending that it be retrieved using the get_job_error_info method
Add test
# Testing Methods Performed
CI

### CODE REVIEW

File: Config/ComputationContainer/Compute1/.env [hunk 1]
```diff
@@ -8,3 +8,4 @@ PARTY_LIST1=http://computation_container1:50020
 PARTY_LIST2=http://computation_container2:50020
 PARTY_LIST3=http://computation_container3:50020
 BTS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb29tX3V1aWQiOiI2NmIzM2RjOC01N2Y1LTQ2NTAtYmM5My02YjdmODVhYjc2ZjYiLCJwYXJ0eV9pbmZvIjpbeyJpZCI6MSwiYWRkcmVzcyI6IjEwLjAuMS4yMCJ9LHsiaWQiOjIsImFkZHJlc3MiOiIxMC4wLjIuMjAifSx7ImlkIjozLCJhZGRyZXNzIjoiMTAuMC4zLjIwIn1dLCJpc3MiOiJRdWlja01QQy1CVFMiLCJzdWIiOiIxMmEyNjQ0Ny0zODRkLTQzMzEtOTY1OS05MWQ3ODg2ODY0YmUiLCJhdWQiOlsiaHR0cDovL2xvY2FsaG9zdDo5MDAxIiwiaHR0cDovL2xvY2FsaG9zdDo5MDAyIiwiaHR0cDovL2xvY2FsaG9zdDo5MDAzIl0sImV4cCI6OTIyMzM3MTk3NDcxOTE3ODc1Mn0.-tnGOijC-GgXwozsN-6pwXRIo60w-j7gHQrg9fANpAk
+TIME_LIMIT=50
```
Comment (KotaTakahashi9320) [inline Config/ComputationContainer/Compute1/.env:11]: 細かい名前

File: src/ManageContainer/Server/LibcToManageContainer/L2m_test.go [hunk 2]
```diff
@@ -165,6 +168,26 @@ func TestGetComputationResult(t *testing.T) {
...
+		JobUuid: "id",
+		Token:   "token_dep",
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if result.GetJobErrorInfo().GetWhat() != "test" {
+		t.Fatal("GetElapsedTime Failed")
```
Comment (KotaTakahashi9320) [inline src/ManageContainer/Server/LibcToManageContainer/L2m_test.go:187]:
```suggestion
		t.Fatal("GetJobErrorInfo Failed")
```

File: src/Proto/LibcToManageContainer/libc_to_manage.proto [hunk 3]
```diff
@@ -185,3 +186,13 @@ message GetElapsedTimeResponse {
...
     double elapsed_time = 2;
 }
+
+message GetJobErrorInfoRequest {
+    string job_uuid = 1;
+    string token = 2;
+}
+
+message GetJobErrorInfoResponse {
+    pb_common_types.JobErrorInfo job_error_info = 1;
```
Comment (arukuka) [inline src/Proto/LibcToManageContainer/libc_to_manage.proto:196]:
```suggestion
    optional pb_common_types.JobErrorInfo job_error_info = 1;
```

File: Config/ComputationContainer/Compute2/.src.env [hunk 4]
```diff
@@ -8,3 +8,4 @@ PARTY_LIST1=http://computation_container1:50120
 PARTY_LIST1=http://computation_container2:50120
 PARTY_LIST3=http://computation_container3:50120
 BTS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb29tX3V1aWQiOiI2NmIzM2RjOC01N2Y1LTQ2NTAtYmM5My02YjdmODVhYjc2ZjYiLCJwYXJ0eV9pbmZvIjpbeyJpZCI6MSwiYWRkcmVzcyI6IjEwLjAuMS4yMCJ9LHsiaWQiOjIsImFkZHJlc3MiOiIxMC4wLjIuMjAifSx7ImlkIjozLCJhZGRyZXNzIjoiMTAuMC4zLjIwIn1dLCJpc3MiOiJRdWlja01QQy1CVFMiLCJzdWIiOiIxMmEyNjQ0Ny0zODRkLTQzMzEtOTY1OS05MWQ3ODg2ODY0YmUiLCJhdWQiOlsiaHR0cDovL2xvY2FsaG9zdDo5MDAxIiwiaHR0cDovL2xvY2FsaG9zdDo5MDAyIiwiaHR0cDovL2xvY2FsaG9zdDo5MDAzIl0sImV4cCI6OTIyMzM3MTk3NDcxOTE3ODc1Mn0.-tnGOijC-GgXwozsN-6pwXRIo60w-j7gHQrg9fANpAk
+TIME_LIMIT=10
```
Comment (arukuka) [inline Config/ComputationContainer/Compute2/.src.env:11]:
```suggestion
TIME_LIMIT=10

```

File: src/ManageContainer/Client/ManageToDb/Client.go [hunk 5]
```diff
@@ -188,6 +190,23 @@ func getComputationStatus(path string) (pb_types.JobStatus, *pb_types.JobErrorIn
...
+
+	_, errInfo, errStatus := getComputationStatus(path)
+	if errStatus != nil {
+		return nil, errStatus
+	}
+	if errInfo != nil {
+		return errInfo, nil
+	}
+
+	return &pb_types.JobErrorInfo{}, nil
```
Comment (arukuka) [inline src/ManageContainer/Client/ManageToDb/Client.go:207]:
```suggestion
	return nil, nil
```

**Review by arukuka — CHANGES_REQUESTED** [review]

File: src/ComputationContainer/ConfigParse/ConfigParse.hpp [hunk 6]
```diff
@@ -88,6 +88,7 @@ class Config
         party_id = getEnvInt("PARTY_ID");
         n_parties = getEnvInt("N_PARTIES");
         sp_id = getEnvInt("SP_ID");
+        time_limit = getEnvInt("TIME_LIMIT");
```
Comment (KotaTakahashi9320) [inline src/ComputationContainer/ConfigParse/ConfigParse.hpp:91]: getEnvOptionalString

File: src/Proto/LibcToManageContainer/libc_to_manage.proto (hunk 3 above)
Comment (KotaTakahashi9320) [reply src/Proto/LibcToManageContainer/libc_to_manage.proto]: @arukuka 
Corrected with a [5a7aee7](https://github.com/acompany-develop/QuickMPC/pull/100/commits/5a7aee75694f88f011a8e794e60218adedd66f23).

File: src/ManageContainer/Client/ManageToDb/Client.go (hunk 5 above)
Comment (KotaTakahashi9320) [reply src/ManageContainer/Client/ManageToDb/Client.go]: Corrected with a [cf193dc](https://github.com/acompany-develop/QuickMPC/pull/100/commits/cf193dcd7535268822e062dcdf0a3fb1ecfbb700)

**Comment by KotaTakahashi9320** [comment]
@arukuka @mdonaka 
Thanks for the review!
Please check the correction made.

**Review by mdonaka — APPROVED** [review]
Sorry for the delay... Thanks for the changes!

**Review by arukuka — APPROVED** [review]
Sorry for late response.  
Thank you!

File: src/ManageContainer/Client/ManageToDb/Client.go (hunk 5 above)
Comment (arukuka) [reply src/ManageContainer/Client/ManageToDb/Client.go]: Thank you!

**Pull request merged**